src/openrct2-ui/windows/NewRide.cpp

```diff
@@ -1056,27 +1057,20 @@ static void window_new_ride_select(rct_window* w)
     ride_construct_new(item);
 }
 
-static void window_new_ride_list_vehicles_for(const uint8_t rideType, const rct_ride_entry* rideEntry, char* out)
+static void window_new_ride_list_vehicles_for(uint8_t rideType, const rct_ride_entry* rideEntry, char* buffer, size_t bufferLen)
 {
-    rct_ride_entry* currentRideEntry;
-    const RideGroup *rideGroup, *currentRideGroup;
-    int32_t rideEntryIndex;
-
+    std::fill_n(buffer, bufferLen, 0);
```

Comment:
Do we need to?

Reply:
I decided the code was simpler doing this at the beginning than first checking for a buffer length >= 0 before putting a null terminator on the first return.

This way is just simpler and safer. The routine isn't called often so it isn't much of a performance issue.